### ISSUE TITLE
Text selection highlight should be greyed out when window loses focus

### ISSUE THREAD

**Issue by Renzo-Olivares** [flutter/flutter#103446]
When a Flutter window loses focus on desktop, the text selection highlight color should be grey.

## Steps to Reproduce

1. Open a Flutter App with a text field.
2. Type in text field and select some text.
3. Open a native application with text input/selection.
4. Type some text in the new window.
5. Select some text in native app
6. Switch focus from native app to flutter app, and back and forth to see the selection highlight color change depending on focus. It only changes on the native app, not the flutter app.

Flutter App on the left (Actual), MacOS Notes App on the right (Expected).

https://user-images.githubusercontent.com/948037/167739336-6bc355e2-6e35-4a54-b024-419199e68c50.mov

<details>
<summary>Code sample</summary>
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});
  
  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: const Center(
        child: TextField(),
      ),
    );
  }
}
```

</details>

**Comment by Renzo-Olivares** [comment]
cc @justinmc

**Comment by justinmc** [comment]
@goderbauer Do we have any method channel that informs the framework if the app was backgrounded or that kind of status?  If not, do you have any opinions on us writing one for this?

**Comment by goderbauer** [comment]
We have https://master-api.flutter.dev/flutter/scheduler/SchedulerBinding/lifecycleState.html, which provides similar information on mobile devices. I am not sure if we have this wired up in the desktop embedder to give the correct status; we should check.

If we need a new API for this, we should talk about how this would work in a multi-window world. Presumably, we'd need this status for each individual window. This probably also needs to feed into the discussion of how to adapt Flutter's focus system in general in the multi window world...

**Comment by justinmc** [comment]
@gspencergoog Will your work on application lifecycle events allow us to handle this?  We'd need to be informed whenever an individual window loses focus.

**Comment by gspencergoog** [comment]
No, it will inform you when the entire application loses focus, but we'll have to wait for the windowing API to tell us when individual windows lose focus.

**Comment by justinmc** [comment]
I think we do have this kind of lifecycle information on desktop nowadays? CC @Renzo-Olivares

**Comment by g0rdan** [comment]
@justinmc @Renzo-Olivares  I'd like to work on it but before I start there is a new behavior rn:

https://github.com/user-attachments/assets/c8c04b79-1312-4438-8c05-55d51bc9e0fe

It does not fix the issue but it indicates "unfocused" state of the selected text. Do yo still think it is worth working on it?

**Comment by Renzo-Olivares** [comment]
@g0rdan, I still think this is worth fixing for the platforms that indicate an unfocused state for their selection like macOS.
